unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,217 @@
+                = {"unregister": {"transaction_id": f"{TRANSACTION_ID}"}}
+            redis_client_auto_register._on_unregister(message)
+
+    def test_on_event_or_response(self, redis_client_auto_register):
+        """assigning the correct arguments to blocking command response and
+        checking by calling on_event_or_response function with correct message"""
+        message = {"data": {"device_uuid": TRANSACTION_ID,
+                            "transaction_id": TRANSACTION_ID}}
+        message["data"] = json.dumps(message["data"])
+        redis_client_auto_register._on_event_or_response(message)
+
+    def test_select_aggregator(self, redis_client_auto_register):
+        """checks if redis_db.publish is called with correct arguments"""
+        aggregator_uuid = str(uuid.uuid4())
+        self.redis_client = redis_client_auto_register
+        self.redis_client.redis_db.publish = MagicMock()
+        self.redis_client.area_uuid = AREA_ID
+        self.redis_client.select_aggregator(aggregator_uuid, is_blocking=False)
+        transaction_id = self.redis_client._transaction_id_buffer[0]
+        data = {"aggregator_uuid": aggregator_uuid,
+                "device_uuid": AREA_ID,
+                "type": "SELECT",
+                "transaction_id": transaction_id}
+        self.redis_client.redis_db.publish.assert_called_once_with("aggregator", json.dumps(data))
+
+    def test_select_aggregator_self_area_uuid_none_throws_exception(self, redis_client_auto_register):
+        """checks if select aggregator with area_uuid is none throws exception"""
+        aggregator_uuid = str(uuid.uuid4())
+        with pytest.raises(RedisAPIException,
+                           match="The device/market has not ben registered yet,"
+                                 " can not select an aggregator"):
+            redis_client_auto_register\
+                .select_aggregator(aggregator_uuid=aggregator_uuid, is_blocking=True)
+
+    def test_select_aggregator_self_area_uuid_not_none_throws_exception(self, redis_client_auto_register):
+        """checks if select aggregator with area_uuid is not none throws exception"""
+        aggregator_uuid = str(uuid.uuid4())
+        with pytest.raises(RedisAPIException,
+                           match="API has timed out."):
+            self.redis_client = redis_client_auto_register
+            self.redis_client.area_uuid = str(uuid.uuid4())
+            redis_client_auto_register\
+                .select_aggregator(aggregator_uuid=aggregator_uuid, is_blocking=True)
+
+    def test_unselect_aggregator_throws_exception(self, redis_client_auto_register):
+        """checks if unselect aggregator throws exception"""
+        aggregator_uuid = str(uuid.uuid4())
+        with pytest.raises(NotImplementedError,
+                           match="unselect_aggregator hasn\'t been implemented yet."):
+            redis_client_auto_register.unselect_aggregator(aggregator_uuid=aggregator_uuid)
```

Comment:
IMO you can skip this. I is not used and will eventually be implemented.

Reply:
Sure